### PR TITLE
Use composition instead of inheritance + Create migrations via injector

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #32 

Based on #74

### CODE REVIEW

**Comment by samdark** [comment]
@vjik please resolve conflicts.

**Comment by vjik** [comment]
> @vjik please resolve conflicts.

Done

File: src/MigrationInterface.php [hunk 1]
```diff
@@ -22,15 +22,8 @@ interface MigrationInterface
...
-    /**
-     * @param bool $value indicates whether the console output should be compacted.
-     *
-     * If this is set to true, the individual commands ran within the migration will not be output to the console.
      *
-     * Default is false, in other words the output is fully verbose by default.
+     * @param MigrationHelper $m
      */
-    public function compact(bool $value): void;
+    public function up(MigrationHelper $m): void;
```
Comment (samdark) [inline src/MigrationInterface.php:28]: Is the idea to execute a single migration multiple times with multiple differently configured `MigrationHelper`s?

File: src/MigrationHelper.php [hunk 2]
```diff
@@ -13,59 +13,32 @@
...
- * For example, the {@see insert()} method can be used to easily insert a row of data into a database table; the
- * {@see createTable()} method can be used to create a database table.
- *
- * Compared with the same methods in {@see Command}, these methods will display extra information showing the method
- * parameters and execution time, which may be useful when applying migrations.
- *
- * For more details and usage information on Migration, see the [guide article on Migration](guide:db-migrations).
- */
-abstract class Migration implements MigrationInterface
+final class MigrationHelper
```
Comment (samdark) [inline src/MigrationHelper.php:16]: May not be the best name. Helpers are usually static classes.

File: src/MigrationInterface.php (hunk 1 above)
Comment (vjik) [reply src/MigrationInterface.php:28]: hm.. maybe for several databases?

File: src/MigrationHelper.php (hunk 2 above)
Comment (vjik) [reply src/MigrationHelper.php:16]: `DbMigrator $db`
`Migrator $m`
?

File: src/MigrationInterface.php (hunk 1 above)
Comment (samdark) [reply src/MigrationInterface.php:28]: Usually it's not the case. I guess @SamMousa meant not making syntax longer and that's it.

File: src/MigrationHelper.php (hunk 2 above)
Comment (samdark) [reply src/MigrationHelper.php:16]: Migrator sounds like something that applies migrations.

Comment (vjik) [reply src/MigrationHelper.php:16]: `Db`
`MigrationQuery`
?

Comment (rustamwin) [reply src/MigrationHelper.php:16]: `MigrationBuilder`?

Comment (samdark) [reply src/MigrationHelper.php:16]: `MigrationBuilder` soulds alright. @vjik let's apply it.

Comment (vjik) [reply src/MigrationHelper.php:16]: > `MigrationBuilder` soulds alright. @vjik let's apply it.

Which variable user in migrations?
I like `$b`, but can `$mb`...

Comment (samdark) [reply src/MigrationHelper.php:16]: `$b` is OK.

Comment (vjik) [reply src/MigrationHelper.php]: Done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**